src/main/java/com/cinema/dao/UserDao.java

```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    Optional<User> findById(Long id);
```

Comment:
you call optional.get() in your controllers so many times

maybe let's return User here?